packages/castmill/lib/castmill/addons/playlists/components/index.tsx

```diff
@@ -104,13 +108,84 @@ const PlaylistsPage: Component<{
       icon: AiOutlineDelete,
       handler: (item: JsonPlaylist) => {
         setCurrentPlaylist(item);
+        setShowConfirmDialog(item);
       },
       label: 'Remove',
     },
-  ] as TableAction<JsonPlaylist>[];
+  ];
+
+  /** It may be possible to refactor this code as most views will have the same UI for
+   * removing resources.
+   */
+  const [showConfirmDialog, setShowConfirmDialog] = createSignal<
+    JsonPlaylist | undefined
+  >();
+  const [showConfirmDialogMultiple, setShowConfirmDialogMultiple] =
+    createSignal(false);
+
+  const confirmRemoveResource = async (resource: JsonPlaylist | undefined) => {
+    if (!resource) {
+      return;
+    }
+    try {
+      await PlaylistsService.removePlaylist(
+        props.store.env.baseUrl,
+        props.store.organizations.selectedId,
+        resource.id
+      );
+
+      refreshData();
+    } catch (error) {
+      alert(`Error removing playlist ${resource.name}: ${error}`);
```

Comment:
maybe something for a later PR but we might want to use something else than `alert` to report errors to the user

Reply:
In the beginning I added these alerts as placeholders, but now I am not so sure that they are that bad actually. At the end of the day it is a built-in mechanism of the browser that kind of works well, it's just maybe not so beautiful... something to think more about.